dataans/src-tauri/src/dialog.rs

```diff
@@ -0,0 +1,27 @@
+use futures::channel::oneshot;
+use std::path::PathBuf;
+use tauri::{command, AppHandle};
+use tauri_plugin_dialog::{DialogExt, FilePath};
+
+#[command]
+pub async fn open_file_dialog(app: AppHandle) -> Result<Option<PathBuf>, String> {
+    // Create a oneshot channel to receive the result
+    let (tx, rx) = oneshot::channel();
+
+    // Spawn a task to open the dialog
+    tauri::async_runtime::spawn(async move {
+        app.dialog()
+            .file()
+            .add_filter("Notes", &["json", "md"])
+            .pick_file(move |file_path| {
+                let path = file_path.and_then(|fp| match fp {
+                    FilePath::Path(p) => Some(p),
+                    _ => None,
```

Comment:
let's log an error here and return it. we do not support importing by URL, so there is no point in using `None`